test/caf_teams_test.f90

```diff
@@ -0,0 +1,62 @@
+module caf_teams_test
+    use iso_c_binding, only: c_size_t, c_ptr, c_intmax_t, c_null_funptr
+    use prif, only: &
+            prif_coarray_handle, &
+            prif_allocate_coarray, &
+            prif_deallocate_coarray, &
+            prif_this_image_no_coarray, &
+            prif_num_images, &
+            prif_team_type, &
+            prif_form_team, &
+            prif_change_team, &
+            prif_end_team
+    use veggies, only: result_t, test_item_t, describe, it, succeed
+
+    implicit none
+    private
+    public :: test_caf_teams
+contains
+    function test_caf_teams() result(tests)
+        type(test_item_t) :: tests
+
+        tests = describe( &
+            "Teams", &
+            [ it("can be created, changed to, and allocate coarrays", check_teams) &
+            ])
+    end function
+
```

Comment:
Ideally tests for different features are separate tests. Admittedly such a test involves the interaction of different features. It may be worth a brief conversation on our testing strategy/organisation.

Reply:
I agree it's a judgement call. 

In any case this topic is clearly orthogonal the purpose this PR, as those queries are not even implemented yet.

Let's not hold up this PR for this.